### PR TITLE
HPCC-23961 Refactor stats to use CRuntimeSummaryStatisticCollection

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

NB: kept in 3 commits for how.
I have more commits to convert other Thor stats to use CThorStatsCollection, but will do in separate PR's.

File: thorlcr/graph/thgraphmaster.ipp [hunk 1]
```diff
@@ -71,43 +71,29 @@ protected:
 
 class graphmaster_decl CThorStatsCollection : public CInterface
 {
+    std::vector<OwnedMalloc<CRuntimeStatisticCollection>> nodeStats;
+    const StatisticsMapping & mapping;
+    CRuntimeSummaryStatisticCollection summary;
```
Comment (ghalliday) [inline thorlcr/graph/thgraphmaster.ipp:76]: This is more naturally a local variable in getStats()

File: thorlcr/graph/thgraphmaster.ipp [hunk 2]
```diff
@@ -71,43 +71,29 @@ protected:
...
-        for (unsigned i=0; i < mapping.numStatistics(); i++)
-            stats[i]->extract(node, source);
+        nodeStats[node]->deserialize(mb);
     }
 
     void getStats(IStatisticGatherer & result)
     {
+        for (unsigned n=0; n < nodeStats.size(); n++) // NB: size is = queryClusterWidth()
+            summary.merge(*nodeStats[n], n);
         for (unsigned i=0; i < mapping.numStatistics(); i++)
```
Comment (ghalliday) [inline thorlcr/graph/thgraphmaster.ipp:96]: I don't think this loop is needed any more

File: thorlcr/graph/thgraphmaster.ipp (hunk 1 above)
Comment (jakesmith) [reply thorlcr/graph/thgraphmaster.ipp:76]: yes, and it's the cause of the x3 anomaly (or how ever many updates there are)

File: thorlcr/graph/thgraphmaster.ipp (hunk 2 above)
Comment (jakesmith) [reply thorlcr/graph/thgraphmaster.ipp:96]: yes, will remove.

**Comment by jakesmith** [comment]
@ghalliday - please see above fixes.

**Pull request merged**